src/libraries/System.Security.Cryptography.X509Certificates/tests/PublicKeyTests.cs

```diff
@@ -12,6 +12,21 @@ namespace System.Security.Cryptography.X509Certificates.Tests
 {
     public static class PublicKeyTests
     {
+        public static bool SupportsBrainpool { get; } = IsCurveSupported(ECCurve.NamedCurves.brainpoolP160r1.Oid);
```

Comment:
Given that macOS doesn't support brainpool curves, I don't think this should be needed.  Any tests that are failing without this indicate a behavioral difference in the lack-of-support between macOS and Android, and that gap should (ideally) be closed.

Reply:
The problem here is that Android can successfully parse a Brainpool X509 key, but it can't actually do anything with said key since the algorithm is unsupported.

Reply:
I'm still not seeing why the test should not run.

* If the GetECDsaPublicKey is throwing a CryptographicException, it'll already pass.
* If the GetECDsaPublicKey is throwing a PNSE, is that consistent?  (It might be that no other platform gets there on unknown/unsupported curves)
  * Then the test can just accept a PNSE in addition to accepting a CryptographicException
* If the GetECDsaPublicKey is succeeding but always returns false on verify, that seems like bad behavior.

Reply:
The third case is what's happening. The public key is valid, so BoringSSL succeeds as parsing the public key and creating the object. However, since the curve is not supported, it later fails when actually trying to use the key during verify.

This also only seems to happen on newer Android API versions. On API Level 21, it correctly throws CryptographicException when parsing the public key.

Reply:
That... seems... odd.

I guess the detector code here realizes there's a failure because the operation is keygen instead of import?

Rather than lose our exception test for Win7 and macOS, I'd prefer something tactical in the test.  Like

```C#
bool isSignatureValid = ...;

if (!isSignatureValid)
{
    // Some versions of Android will import this key without reporting that it's for an unknown curve,
    // we first learn about the problem when all verification operations fail.
    Assert.True(Whatever.IsAndroid());
    return;
}

unchecked
{
   etc
}
```